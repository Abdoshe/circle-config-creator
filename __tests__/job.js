/* @flow */

import Job from '../src/job';
import Docker from '../src/executors/docker';

const docker = new Docker('test-image:latest');

describe('Job', () => {
  it('can create a simple job', () => {
    const job = new Job('test_job').executor(docker);
    expect(job.compose()).toMatchSnapshot();
  });

  it('complains if no executor', () => {
    const job = new Job('test_job');
    expect(() => job.compose()).toThrowErrorMatchingSnapshot();
  });

  it('makes a more complex job', () => {
    const job = new Job('test_job')
      .executor(docker)
      .checkout()
      .run('echo "hello, world"')
      .run({ command: 'echo "Job is over"', workingDirectory: '~/project' });

    expect(job.compose()).toMatchSnapshot();
  });

  it('can progressively restore a cache', () => {
    const job = new Job('test-job')
      .executor(docker)
      .progressiveRestoreCache('v1-this-is-a-test');

    expect(job.compose()).toMatchSnapshot();
  });

  it('can restore a cache progressively with a base', () => {
    const job = new Job('test-job')
      .executor(docker)
      .progressiveRestoreCache('v1-this-is-a-test', 'v1-this-is');

    expect(job.compose()).toMatchSnapshot();
  });

  it('can restore a cache progressively with no base (first item)', () => {
    const job = new Job('test-job')
      .executor(docker)
      .progressiveRestoreCache('v1-this-is-a-test', '');

    expect(job.compose()).toMatchSnapshot();
  });

  it('can be updated immutably', () => {
    const jobOne = new Job('test-job');
    const jobTwo = jobOne.progressiveRestoreCache('v1-this-is-a-test');

    expect(jobOne).not.toEqual(jobTwo);
  });
});
