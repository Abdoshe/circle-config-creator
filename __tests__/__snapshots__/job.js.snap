// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job can create a simple job 1`] = `
Object {
  "test_job": Object {
    "docker": Array [
      Object {
        "image": "test-image:latest",
      },
    ],
    "steps": Array [],
  },
}
`;

exports[`Job can progressively restore a cache 1`] = `
Object {
  "test-job": Object {
    "docker": Array [
      Object {
        "image": "test-image:latest",
      },
    ],
    "steps": Array [
      Object {
        "restore_cache": Object {
          "keys": Array [
            "v1-this-is-a-test",
            "v1-this-is-a-",
            "v1-this-is-",
            "v1-this-",
          ],
          "name": "Restoring Cache",
        },
      },
    ],
  },
}
`;

exports[`Job can restore a cache progressively with a base 1`] = `
Object {
  "test-job": Object {
    "docker": Array [
      Object {
        "image": "test-image:latest",
      },
    ],
    "steps": Array [
      Object {
        "restore_cache": Object {
          "keys": Array [
            "v1-this-is-a-test",
            "v1-this-is-a-",
            "v1-this-is-",
          ],
          "name": "Restoring Cache",
        },
      },
    ],
  },
}
`;

exports[`Job can restore a cache progressively with no base (first item) 1`] = `
Object {
  "test-job": Object {
    "docker": Array [
      Object {
        "image": "test-image:latest",
      },
    ],
    "steps": Array [
      Object {
        "restore_cache": Object {
          "keys": Array [
            "v1-this-is-a-test",
            "v1-this-is-a-",
            "v1-this-is-",
            "v1-this-",
            "v1-",
          ],
          "name": "Restoring Cache",
        },
      },
    ],
  },
}
`;

exports[`Job complains if no executor 1`] = `"You must set an executor for \`test_job\`"`;

exports[`Job makes a more complex job 1`] = `
Object {
  "test_job": Object {
    "docker": Array [
      Object {
        "image": "test-image:latest",
      },
    ],
    "steps": Array [
      "checkout",
      Object {
        "run": "echo \\"hello, world\\"",
      },
      Object {
        "run": Object {
          "command": "echo \\"Job is over\\"",
          "no_output_timeout": undefined,
          "working_directory": "~/project",
        },
      },
    ],
  },
}
`;
